### ISSUE TITLE
Save document does not prompt invalid JSON

### ISSUE THREAD

**Issue by linquize** [apache/couchdb-fauxton#954]
Save document does not prompt invalid JSON

## Expected Behavior
Prompt the user "Invalid JSON"

## Current Behavior
1. If **Save** button is clicked, nothing shown. In browser console, a Javascript error occurred.
```
SyntaxError: JSON.parse: expected ',' or '}' after property value in object at line 9 column 5 of the JSON data
```
2. If **ctrl+S** is pressed, the browser's Save File Dialog shows, which is not the expected behaviour

## Possible Solution
Prompt "invalid JSON", and handle **ctrl+S** hotkey when error

## Steps to Reproduce (for bugs)
1. Edit a document with invalid JSON format
2. Click the green "Save" button / Press hotkey **Ctrl+S**

## Context
Does not realize I have JSON typo

## Your Environment
* Version used: CouchDB 2.1.0
* Browser Name and version: Firefox 54 x64
* Operating System and version (desktop or mobile): Win 10 x64 version 1703
* Link to your project: N/A

**Comment by linquize** [comment]
Ping!

**Comment by Antonio-Maranhao** [comment]
The error message is displayed on the latest version in master using Firefox 56 (64bit) on Mac.
Ctrl+S also triggers the save operation.

![image](https://user-images.githubusercontent.com/30349380/31095505-0d775284-a787-11e7-8def-0db69558b6ae.png)

**Comment by linquize** [comment]
Still an issue in couchdb 2.1.1

**Comment by popojargo** [comment]
This is fixed on the latest version of the master. The 2.1.1 build probably used an older tag of Fauxton. 

Proof: 

![image](https://user-images.githubusercontent.com/9197078/40886975-50a4bae0-670f-11e8-8f63-1228888c9278.png)

@Antonio-Maranhao This should be closed.

**Comment by wohali** [comment]
Will be fixed in the 2.2.0 release.
